### PR TITLE
listen for signals and exit cleanly

### PR DESCRIPTION
**Pull request by maplebed**
Finish up what we were doing and exit nicely, but force quit if it takes longer than 10sec.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -26,11 +30,28 @@ func main() {
 		log.Fatal(err)
 	}
 
+	sigs := make(chan os.Signal, 1)
+	abort := make(chan bool, 1)
+	signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM)
+	go func() {
+		sig := <-sigs
+		io.WriteString(os.Stderr, fmt.Sprintf("Aborting! Caught signal \"%s\"\n", sig))
```
Comment (christineyen) [inline main.go]: i suspect you may want the simpler

`fmt.Fprintf(os.Stderr, "Aborting! Caught Signal \"%s\"\n", sig))`

(ditto down below, `fmt.Fprint(os.Stderr, "Cleaning up...\n")`

File: main.go [hunk 2]
```diff
@@ -26,11 +30,28 @@ func main() {
...
+		io.WriteString(os.Stderr, "Cleaning up...\n")
+		abort <- true
+		go func() {
```
Comment (christineyen) [inline main.go]: something about this goroutine immediately inside another goroutine smells.

does something like:

```go
select {
  case abort <- true:
  case <- time.After(10 * time.Second):
    fmt.Fprintf(os.Stderr, "Taking too long....")
    os.Exit(1)
}
```

work just as well?

Comment (christineyen) [inline main.go]: relatedly - is 10 seconds acceptable to your non-Honeycomber opsen persona?

I guess so long as we're outputting somewhere the fact that we're **trying** to exit, 10s isn't too painfully long.

File: main.go (hunk 1 above)
Comment (maplebed) [inline main.go]: tnx, done.

File: main.go (hunk 2 above)
Comment (maplebed) [inline main.go]: done.

**Pull request merged**